marconi/test/MintBurn.hs

```diff
@@ -0,0 +1,400 @@
+{-# LANGUAGE OverloadedStrings  #-}
+{-# LANGUAGE TemplateHaskell    #-}
+module MintBurn where
+
+import Codec.Serialise (serialise)
+import Control.Concurrent qualified as IO
+import Control.Concurrent.Async qualified as IO
+import Control.Concurrent.STM qualified as IO
+import Control.Exception (catch)
+import Control.Lens qualified as Lens
+import Control.Monad (foldM, forM, replicateM, void, when)
+import Control.Monad.IO.Class (liftIO)
+import Data.ByteString.Lazy qualified as LBS
+import Data.ByteString.Short qualified as SBS
+import Data.Coerce (coerce)
+import Data.List.NonEmpty (NonEmpty ((:|)))
+import Data.List.NonEmpty qualified as NonEmpty
+import Data.Map qualified as Map
+import Data.Maybe (mapMaybe)
+import Data.Set qualified as Set
+import Data.String (fromString)
+import Data.Word (Word64)
+import Prettyprinter (defaultLayoutOptions, layoutPretty, pretty, (<+>))
+import Prettyprinter.Render.Text (renderStrict)
+import Streaming.Prelude qualified as S
+import System.Directory qualified as IO
+import System.FilePath ((</>))
+
+import Hedgehog (Gen, Property, forAll, (===))
+import Hedgehog qualified as H
+import Hedgehog.Extras.Test qualified as HE
+import Hedgehog.Extras.Test.Base qualified as H
+import Hedgehog.Gen qualified as Gen
+import Hedgehog.Range qualified as Range
+import Test.Tasty (TestTree, testGroup)
+import Test.Tasty.Hedgehog (testPropertyNamed)
+
+import Cardano.Api qualified as C
+import Cardano.Api.Shelley qualified as C
+import Cardano.BM.Setup (withTrace)
+import Cardano.BM.Trace (logError)
+import Cardano.BM.Tracing (defaultConfigStdout)
+import Cardano.Streaming (ChainSyncEventException (NoIntersectionFound), withChainSyncEventStream)
+import Gen.Cardano.Api.Typed qualified as CGen
+import Plutus.V1.Ledger.Api qualified as PlutusV1
+import PlutusTx qualified
+import Test.Base qualified as H
+import Testnet.Cardano qualified as TN
+
+import Helpers qualified as TN
+import Marconi.Index.MintBurn qualified as MintBurn
+import Marconi.Indexers qualified as M
+import Marconi.Logging ()
+import RewindableIndex.Storable qualified as RI
+
+-- | Each test case is described beside every top level property
+-- declaration.
+tests :: TestTree
+tests = testGroup "MintBurn"
+  [ testPropertyNamed "mintsPreserved" "mintsPreserved" mintsPreserved
+  , testPropertyNamed "queryMintedValues" "queryMintedValues" queryMintedValues
+  , testPropertyNamed "resume" "resume" resume
+  , testPropertyNamed "rewind" "rewind" rewind
+  , testPropertyNamed "intervals" "intervals" intervals
+  , testPropertyNamed "endToEnd" "endToEnd" endToEnd
+  ]
+
+-- | This is a sanity-check test that turns a TxBodyContent with mint
+-- events into a TxBody and checks if the mint events are found in the
+-- result. It doesn't test an indexer.
+--
+-- Create a transaction that mints a value, convertit to a @Tx@
+-- through `makeTransactionBody` and `signShelleyTransaction` and find
+-- that the minted assets are still there.
```

Comment:
Maybe reword this? The two paragraphs seem to repeat the same thing but in different terms.